src/transformers/models/whisper/modeling_whisper.py

```diff
@@ -1608,6 +1615,18 @@ def generate(
         if len(forced_decoder_ids) > 0:
             generation_config.forced_decoder_ids = forced_decoder_ids
 
+        if initial_prompt_ids is not None:
+            if kwargs.get("decoder_start_token_id") is not None:
+                raise ValueError(
+                    "When specifying `initial_prompt_ids`, you cannot also specify `decoder_start_token_id` as it gets overwritten."
+                )
+            # Set <|startofprev|> to the decoder_start_token_id
+            kwargs.update({"decoder_start_token_id": initial_prompt_ids.pop(0)})
+            initial_prompt_ids.append(generation_config.decoder_start_token_id)
+            initial_prompt_ids.extend([id for _, id in generation_config.forced_decoder_ids])
+            indexed_initial_prompt_ids = [[idx + 1, id] for idx, id in enumerate(initial_prompt_ids)]
```

Comment:
Nitpick: The `forced_decoder_ids` use a tuple instead of a list, so this should be a tuple too:

```python
indexed_initial_prompt_ids = [(rank + 1, token) for rank, token in enumerate(initial_prompt_ids)]
```
I also used `rank` instead of `idx` and `token` instead of `id`, to be consistent with `tokenizer.get_decoder_prompt_ids`.

Reply:
I'm also not 100% sure that `generation_config.forced_decoder_ids` is always filled in here. For example, if we do the following:

```python
forced_decoder_ids = processor.get_decoder_prompt_ids(language="de", task="translate")
predicted_ids = model.generate(input_features, forced_decoder_ids=forced_decoder_ids)
```

then `generation_config.forced_decoder_ids` may not have the `<|de|>` token at this point. In this situation, the correct `forced_decoder_ids` will be filled in by `super().generate(...)`, as far as I can tell.

Reply:
Ah good catch! Yes before the implementation was overriding provided forced_decoder_ids, but this should be fixed now. Before it was adding these from the generation_config only, but now checks kwargs first.